From 723649a6b024ddf4d8c5d47e2510a9ab87ae02e0 Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Wed, 10 Sep 2014 18:29:08 -0400
Subject: [ethernet] ixgbe: remove open-coded skb_cow_head

Message-id: <1410373750-11341-49-git-send-email-jogreene@redhat.com>
Patchwork-id: 93529
O-Subject: [RHEL7.1 PATCH 48/50] ixgbe: remove open-coded skb_cow_head
Bugzilla: 1091123
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

BZ: 1091123
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7919168
Tested: In beaker. See BZ for details

Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
Cc: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Phil Schmitt <phillip.j.schmitt@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit 2049e1f6f5e6720b182c42695dbdef071804226e)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c b/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
index da67562..1305c64 100644
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
@@ -6260,6 +6260,7 @@ static int ixgbe_tso(struct ixgbe_ring *tx_ring,
  struct sk_buff *skb = first->skb;
  u32 vlan_macip_lens, type_tucmd;
  u32 mss_l4len_idx, l4len;
+ int err;
 
  if (skb->ip_summed != CHECKSUM_PARTIAL)
   return 0;
@@ -6267,11 +6268,9 @@ static int ixgbe_tso(struct ixgbe_ring *tx_ring,
  if (!skb_is_gso(skb))
   return 0;
 
- if (skb_header_cloned(skb)) {
-  int err = pskb_expand_head(skb, 0, 0, GFP_ATOMIC);
-  if (err)
-   return err;
- }
+ err = skb_cow_head(skb, 0);
+ if (err < 0)
+  return err;
 
  /* ADV DTYP TUCMD MKRLOC/ISCSIHEDLEN */
  type_tucmd = IXGBE_ADVTXD_TUCMD_L4T_TCP;
@@ -6818,8 +6817,8 @@ netdev_tx_t ixgbe_xmit_frame_ring(struct sk_buff *skb,
      IXGBE_TX_FLAGS_VLAN_PRIO_SHIFT;
   if (tx_flags & IXGBE_TX_FLAGS_SW_VLAN) {
    struct vlan_ethhdr *vhdr;
-   if (skb_header_cloned(skb) &&
-       pskb_expand_head(skb, 0, 0, GFP_ATOMIC))
+
+   if (skb_cow_head(skb, 0))
     goto out_drop;
    vhdr = (struct vlan_ethhdr *)skb->data;
    vhdr->h_vlan_TCI = htons(tx_flags >>
-- 
1.7.1